app/src/page/file/view.js

```diff
@@ -34,15 +36,27 @@ class FilePage extends React.PureComponent {
   
   componentDidMount() {
     StatusBar.setHidden(false);
+    
+    const { isResolvingUri, resolveUri, navigation } = this.props;
+    const { uri } = navigation.state.params;
+    if (!isResolvingUri) resolveUri(uri);
+    
     this.fetchFileInfo(this.props);
     this.fetchCostInfo(this.props);
+
     if (NativeModules.Mixpanel) {
-      NativeModules.Mixpanel.track('Open File Page', { Uri: this.props.navigation.state.params.uri });
+      NativeModules.Mixpanel.track('Open File Page', { Uri: uri });
     }
   }
 
   componentWillReceiveProps(nextProps) {
```

Comment:
This should be moved to componentDidUpdate since componentWillReceiveProps will be deprecated